tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -496,7 +494,11 @@ private:
          void Advance()
          {
             auto fnNext_Contig = [&]() {
-               auto &iter = *static_cast<unsigned char **>(fIterator), p = iter;
+               // When a staging area is used, `fIterator` is a pointer to the first element in memory. However, for
+               // other types of contiguous memory, it will be a pointer to this pointer.
+               auto &iter = fOwner.fUsesStaging ? *static_cast<unsigned char **>((void *)&fIterator)
+                                                : *static_cast<unsigned char **>(fIterator);
```

Comment:
I'll check tomorrow, but I thought the staging area was handled exactly the same way as a kSTLvector collection (@pcanal ?).

Reply:
As far as I know they are the same.  See the implementation of TGenCollectionProxy__StagingCreateIterators and TGenCollectionProxy__VectorCreateIterators in TGenCollectionProxy.cxx.

If this does not make the test fails, can we add some test that exercise this code path .. or is it compensated elsewhere?

Reply:
The inconsistency is covered by the `ntuple_types` test, through `StdSet` on l213 and `TVirtualCollectionProxy` on l1123 (i.e. one of either will fail if we don't address both options). However now that I think about it, it maybe makes more sense to make sure `Func_CreateIterators` from the `SimpleCollectionProxy` consistent with the other functions, although I'm not sure what that would mean for e.g. the `DataVector` support in Athena.

Reply:
Apparently, after a quick read of both _(i)_ the functions mentioned by @pcanal (i.e., `TGenCollectionProxy__(Staging|Vector)CreateIterators`), and _(ii)_ the test code for [SimpleCollectionProxy](https://github.com/root-project/root/blob/master/tree/ntuple/v7/test/SimpleCollectionProxy.hxx#L20), this shouldn't be required at all.

If it is, I think we are doing something wrong, somewhere; perhaps we could have a chat today on this.

For now, I'll submit the review so that the other comments can be addressed.